messaging/src/test/java/org/axonframework/commandhandling/gateway/AbstractRetrySchedulerTest.java

```diff
@@ -0,0 +1,231 @@
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.commandhandling.CommandExecutionException;
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.common.AxonNonTransientException;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Predicate;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+public class AbstractRetrySchedulerTest {
+
+  public static class RetrySchedulerStub extends AbstractRetryScheduler {
+
+    protected RetrySchedulerStub(Builder builder) {
+      super(builder);
+    }
+
+    @SuppressWarnings("rawtypes")
+    @Override
+    protected long computeRetryInterval(CommandMessage commandMessage, RuntimeException lastFailure, List<Class<? extends Throwable>[]> failures) {
+      return 100;
+    }
+
+    static Builder builder() {
+      return new Builder();
+    }
+
+    public static class Builder extends AbstractRetryScheduler.Builder<Builder> {
+      RetrySchedulerStub build() {
+        return new RetrySchedulerStub(this);
+      }
+    }
+  }
+
+  @Test
+  void isExplicitlyNonTransient_defaults() {
+    RetrySchedulerStub retrySchedulerStub = RetrySchedulerStub.builder().retryExecutor(mock(ScheduledExecutorService.class)).build();
+    assertTrue(
+        retrySchedulerStub.isExplicitlyNonTransient(new AxonNonTransientException("message") {}),
+        "AxonNonTransientException should be treated as non-transient by default"
+    );
+  }
+
+  @Test
+  void isExplicitlyNonTransient_defaults_and_addNonTransientFailures() {
+    RetrySchedulerStub retrySchedulerStub = RetrySchedulerStub
+        .builder()
+        .retryExecutor(mock(ScheduledExecutorService.class))
+        .addNonTransientFailurePredicate(new NonTransientClassesPredicate(CommandExecutionException.class, IllegalArgumentException.class))
+        .build();
+
+    assertTrue(
+        retrySchedulerStub.isExplicitlyNonTransient(new AxonNonTransientException("message") {}),
+        "AxonNonTransientException should be treated as non-transient by default"
+    );
+
+    assertTrue(
+        retrySchedulerStub.isExplicitlyNonTransient(new CommandExecutionException("message", null)),
+        "Per configuration, CommandExecutionException should be treated as non-transient"
+    );
+
+    assertTrue(
+        retrySchedulerStub.isExplicitlyNonTransient(new IllegalArgumentException("message", null)),
+        "Per configuration, IllegalArgumentException should be treated as non-transient"
+    );
+  }
+
+  @Test
+  void isExplicitlyNonTransient_nonTransientFailurePredicate_throwable() {
+    Predicate<Throwable> nonTransientFailurePredicate = (Throwable failure) -> "I'm non-transient failure".equals(failure.getMessage());
```

Comment:
Gotcha, now I see why you're not using the typed `Predicate` as the type in the `RetryScheduler`.

How about we introduce a `NonTrasientFailurePredicate` interface implemented by the `NonTransientClassesPredicate` and the default instance?
Let me know what you think.

Reply:
I'm not sure about the interface as I don't see the real value. I mean, `NonTransientClassesPredicate` is just a simple java predicate, it is not hardcoded anywhere, it can be easily replaced with another predicate, and probably there isn't a real need to mock it in tests. 

If you have any particular reason or scenario, can you describe it for me?

Reply:
I'd wager that at this stage, the only value could be to-the-point documentation on the interface, as well as not using the generic method `test` in favor of a declarative name.
Added, named combine methods and/or static (and straightforward) implementations to accept/reject everything could be added to the interface.

Honestly, not a big point of discussion, as introducing it after the fact is straightforward (we did this for snapshot filters, which used to be a `Predicate`, but now is a `SnapshotFilter`.

Reply:
In favor of getting things done, I am moving the discussion to use a dedicated `interface` to describe the predicate as a future problem. Hence, this conversation is resolved.